### ISSUE TITLE
[Bug]: his analysis terminated unexpectedly.  Error in sort.int(sapply(medPaths, length), index.return = TRUE): 'x' must be atomic

### ISSUE THREAD

**Issue by ektas55** [jasp-stats/jasp-issues#3108]
### JASP Version

0.19.2

### JASP Module

Process

### What analysis are you seeing the problem on?

PROCESS moderated mediation

### What OS are you seeing the problem on?

Windows 11

### Bug Description

his analysis terminated unexpectedly.

Error in sort.int(sapply(medPaths, length), index.return = TRUE): 'x' must be atomic
tryCatchOne(expr, names, parentenv, handlers[[1]])

doTryCatch(return(expr), name, parentenv, handler)

withCallingHandlers(expr = analysis(jaspResults = jaspResults, dataset = dataset, options = options), error = .addStackTrace)

analysis(jaspResults = jaspResults, dataset = dataset, options = options)

.procModelSyntax(jaspResults, options)

.procModelSyntaxSingleModel(modelsContainer[[modelName]], modelOptions)

.procMedEffectsSyntax(graph, container[['modProbes']]$object, container[['contrasts']]$object)

sort.int(sapply(medPaths, length), index.return = TRUE)

stop(''x' must be atomic')

To receive assistance with this problem, please report the message above at: https://jasp-stats.org/bug-reports

### Expected Behaviour

It is not recognizing the variables for certain models. It takes only one variable for analysis despite providing multiple variables

### Steps to Reproduce

1.I provided multiple variables
2. When I provided Model7 of PROCESS with two mediators and one moderator, the error report popped up terminating the program
3.
...

### Final Checklist

- [ ] I have included a screenshot showcasing the issue, if possible.
- [ ] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by tomtomme** [comment]
> I think this error occurred in some cases when a variable in the analysis had the Ordinal type.
from https://github.com/jasp-stats/jasp-issues/issues/2666

**Comment by maltelueken** [comment]
Hi, thanks for making the issue. Could you provide a screenshot of the types of variables that you used or provide the JASP file that creates the error?

Currently, not all Hayes models accept multiple mediators (they only take the first one and ignore the rest). Models that do accept multiple mediators are 4, 5, 6, 80, 81. We can add model 7 in the next release.

**Comment by ektas55** [comment]
I shared the file on the mail

**Comment by tomtomme** [comment]
@maltelueken 
Do you have Access to the issue Mailaccount?

**Comment by maltelueken** [comment]
Yes, I received the file (and updated my comment). When I reran the file with version 19.2, the error did not occur.

**Comment by tomtomme** [comment]
@maltelueken 
Understood. Did you test on win11? Could also be OS-specific.

**Comment by ektas55** [comment]
Yes its WIN 11.

**Comment by tomtomme** [comment]
@ektas55 
I asked malte :D

But since you are still online - are you sure that the error persists with 0.19.2 on your end since it seems the file you send was build with an older version. And malte cannot reproduce the error with 0.19.2

**Comment by ektas55** [comment]
I shared the complete message that I was on screen

**Comment by tomtomme** [comment]
I know. But can you Not say which jasp version you used? If you are unsure, please Update jasp by downloading it again and retest please.

**Comment by ektas55** [comment]
it's 0.18.30

**Comment by maltelueken** [comment]
Can you download and install the latest version 19.2 and check if the error still persists?

**Comment by ektas55** [comment]
Thanks

**Comment by tomtomme** [comment]
Does this mean we can close this issue?

**Comment by ektas55** [comment]
I have installed the latest version but it still shows issues... I think
moderated.moderated mediation does not work...

**Comment by tomtomme** [comment]
@ektas55 
thx for the report. Can you give more details on which model exactly does not work and what is the error message? Also uploading the affected jasp-file would be helpful
